app/src/main/res/layout/pin_password_activity.xml

```diff
@@ -42,152 +42,147 @@
+          android:layout_marginTop="12dp"
+          android:text="@{viewModel.profile.isAdmin ? @string/pin_password_admin_enter : @string/pin_password_user_enter}"
+          android:textColor="@color/component_color_shared_secondary_2_text_color"
+          app:layout_constraintEnd_toEndOf="@+id/hello_text_view"
+          app:layout_constraintStart_toStartOf="@+id/hello_text_view"
+          app:layout_constraintTop_toBottomOf="@id/hello_text_view" />
 
         <androidx.constraintlayout.widget.ConstraintLayout
           android:layout_width="@dimen/pin_password_activity_input_with_icon_width"
           android:layout_height="wrap_content"
-          android:layout_gravity="center_horizontal"
-          android:layout_marginStart="@dimen/pin_password_activity_margin_start"
-          android:layout_marginEnd="@dimen/pin_password_activity_margin_end"
-          android:paddingTop="@dimen/pin_password_activity_padding_top">
-
-          <TextView
-            android:id="@+id/hello_text_view"
-            style="@style/Heading1ViewStart"
+          android:layout_marginTop="24dp"
+          app:layout_constraintEnd_toEndOf="parent"
+          app:layout_constraintStart_toStartOf="parent"
+          app:layout_constraintTop_toBottomOf="@+id/enter_text_view">
+
+          <com.google.android.material.textfield.TextInputLayout
+            android:id="@+id/pin_password_input_pin"
+            style="@style/TextInputLayout"
             android:layout_width="0dp"
             android:layout_height="wrap_content"
-            android:layout_marginStart="@dimen/pin_password_activity_hello_text_view_margin_start"
-            android:layout_marginEnd="@dimen/pin_password_activity_hello_text_view_margin_end"
-            android:fontFamily="sans-serif-medium"
-            android:text="@{viewModel.helloText}"
-            android:textColor="@color/component_color_shared_primary_text_color"
-            app:layout_constraintEnd_toEndOf="parent"
-            app:layout_constraintHorizontal_bias="0.0"
+            android:layout_marginTop="2dp"
+            android:hint="@{viewModel.profile.isAdmin ? @string/input_pin_password_as_admin : @string/input_pin_password_as_user}"
+            android:minHeight="56dp"
+            app:errorMessage="@{viewModel.errorMessage}"
+            app:layout_constraintEnd_toStartOf="@id/show_pin"
             app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toTopOf="parent" />
+            app:layout_constraintTop_toTopOf="parent">
+
+            <com.google.android.material.textfield.TextInputEditText
+              android:id="@+id/pin_password_input_pin_edit_text"
+              style="@style/TextInputEditText"
+              android:layout_width="match_parent"
+              android:layout_height="wrap_content"
+              android:imeOptions="actionNext"
+              android:inputType="numberPassword"
+              android:maxLength="@{viewModel.profile.pin.length()}"
```

Comment:
I think since profile is a LiveData, it will be good to add a conditional, something like: @{viewModel.profile != null ? viewModel.profile.pin.length() : 5}, otherwise this risks a null dereference during the pending state.